annotation-processor/src/test/java/org/hibernate/validator/ap/testmodel/overriding/MethodOverridingTests.java

```diff
@@ -0,0 +1,479 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.ap.testmodel.overriding;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.CONSTRUCTOR;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE_USE;
+
+import java.lang.annotation.Target;
+
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+import org.hibernate.validator.constraints.NotBlank;
+
+public class MethodOverridingTests {
+
+	/**
+	 * Case 1: annotation on overridden method parameters - incorrect
+	 */
+	public static class MethodOverridingTestCase1 {
+		public void doSomething(String param) {
+
+		}
+	}
+
+	public static class MethodOverridingTestCase1Sub extends MethodOverridingTestCase1 {
+
+		@Override
+		public void doSomething(@NotNull String param) {
+			super.doSomething( param );
+		}
+	}
+
+	/**
+	 * Case 2: Same set of annotations on overridden method parameters as on the parent - incorrect
+	 */
+	public static class MethodOverridingTestCase2 {
+		public void doSomething(@NotNull @NotBlank @Size(max = 10) String param) {
+
+		}
+	}
+
+	public static class MethodOverridingTestCase2Sub extends MethodOverridingTestCase2 {
+
+		@Override
+		public void doSomething(@NotNull @NotBlank @Size(max = 10) String param) {
```

Comment:
Is there a strong need to check for this? Technically, it doesn't represent any constraint strengthening.